pkg/cli/config/adminkubeconfig/new_admin_kubeconfig.go

```diff
@@ -0,0 +1,267 @@
+package adminkubeconfig
+
+import (
+	"bytes"
+	"context"
+	"crypto/x509"
+	"fmt"
+	"reflect"
+	"time"
+
+	"github.com/openshift/library-go/pkg/crypto"
+	"github.com/openshift/library-go/pkg/operator/certrotation"
+	"github.com/openshift/oc/pkg/cli/config/refreshcabundle"
+	"github.com/spf13/cobra"
+	corev1 "k8s.io/api/core/v1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/tools/clientcmd"
+	clientcmdapi "k8s.io/client-go/tools/clientcmd/api"
+	"k8s.io/client-go/util/cert"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+)
+
+type NewAdminKubeconfigOptions struct {
+	RESTClientGetter genericclioptions.RESTClientGetter
+
+	genericclioptions.IOStreams
+}
+
+var (
+	newAdminKubeconfigLong = templates.LongDesc(i18n.T(`
+		Generate, make the server trust, and display a new admin.kubeconfig.
+
+		The key is produced locally and is not persisted to disk.  The public half is pushed to the cluster
+		for the kube-apiserver to trust the locally created admin.kubeconfig.`))
+
+	newAdminKubeconfigExample = templates.Examples(`
+		# Generate a new admin kubeconfig
+		oc config new-admin-kubeconfig`)
+)
+
+const (
+	adminKubeconfigClientCAConfigMap = "admin-kubeconfig-client-ca"
+
+	tenYears = 24 * time.Hour * 365 * 10
```

Comment:
At some point we'll want to centralize our many "tenYears" constants...

Reply:
> At some point we'll want to centralize our many "tenYears" constants...

Just in case 10 years changes in duration some day? ;)

Reply:
I mean more that I can imagine us wanting to have a more consistent policy for certs, e.g. something much closer to Let's Encrypt expiry at e.g. 3 months to ensure that rotation really works.  To do that we'd want to centralize these currently disparate constants.

Just a side comment, not saying we need to change anything now.